src/language/githubReleaseInstaller.ts

```diff
@@ -121,32 +121,88 @@ export class GitHubReleaseInstaller {
     return this.getConfiguredTagAndVersionCache;
   }
 
+  public static readonly CurrentVersionTag = 'current-version';
+
+  // If there is a major version bump and the version is auto-selected
+  // warn the user before hand
+  private static async dafny4upgradeCheck(
```

Comment:
What's the code duplication for?

Reply:
I wanted to check with you, it looks like this one is the only active place, I thought you had activated the dotnet tool installer, but apparently the class is not used anywhere. Is that intended? I plan to either remove one or reduce duplication.

Reply:
The tool installer isn't active because it doesn't include Z3 and the tool doesn't know how to download Z3.

Reply:
Let me just refactor it for now so that it's not duplicated, but we'll keep it there just in case.